code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -2,6 +2,9 @@
 	set name = "quick-equip"
 	set hidden = TRUE
 
+	if(is_mob_incapacitated() || lying || buckled)
```

Comment:
buckled seems unnecessary, or perhaps I'm missing out the motive.
This should also be undoable whilst inside a mecha, pretty much like other inventory interactions.